contracts/core/OTCWrapper.sol

```diff
@@ -0,0 +1,690 @@
+            address(this), // owner
+            address(this), // address to transfer from
+            order.collateral, // deposited asset
+            order.vaultID, // vaultId
+            _amount, // amount
+            0, //index
+            "" //data
+        );
+
+        // execute actions
+        controller.operate(actions);
+
+        emit CollateralDeposited(_orderID, _amount);
+    }
+
+    /**
+     * @notice allows market maker to withdraw collateral
+     * @dev can only be called by the market maker who is the order seller
+     * @param _orderID id of the order
+     * @param _amount amount to withdraw
+     */
+    function withdrawCollateral(uint256 _orderID, uint256 _amount) external nonReentrant {
+        require(_orderID <= latestOrder, "OTCWrapper: inexistent order");
+
+        Order memory order = orders[_orderID];
+
+        require(order.seller == msg.sender, "OTCWrapper: sender is not the order seller");
+
+        (UtilsWrapperInterface.Vault memory vault, , ) = controller.getVaultWithDetails(address(this), order.vaultID);
+
+        require(
+            marginRequirements.checkWithdrawCollateral(
+                order.seller,
+                order.notional,
+                _amount,
+                order.oToken,
+                order.vaultID,
+                vault
+            ),
+            "OTCWrapper: insufficient collateral"
+        );
+
+        UtilsWrapperInterface.ActionArgs[] memory actions = new UtilsWrapperInterface.ActionArgs[](1);
+
+        actions[0] = UtilsWrapperInterface.ActionArgs(
+            UtilsWrapperInterface.ActionType.WithdrawCollateral,
+            address(this), // owner
+            msg.sender, // address to transfer to
+            order.collateral, // withdrawn asset
+            order.vaultID, // vaultId
+            _amount, // amount
+            0, //index
+            "" //data
+        );
+
+        // execute actions
+        controller.operate(actions);
+
+        emit CollateralWithdrawn(_orderID, _amount);
+    }
+
+    /**
+     * @notice Deposits the `asset` from msg.sender without an approve
+     * `v`, `r` and `s` must be a valid `secp256k1` signature from `owner`
+     * over the EIP712-formatted function arguments
+     * @param _acct signer account
+     * @param _asset is the asset address to deposit
+     * @param _amount is the amount to deposit
+     * @param _deadline must be a timestamp in the future
+     * @param _v is a valid signature
+     * @param _r is a valid signature
+     * @param _s is a valid signature
+     */
+    function _depositWithPermit(
+        address _acct,
+        address _asset,
+        uint256 _amount,
+        uint256 _deadline,
+        uint8 _v,
+        bytes32 _r,
+        bytes32 _s
+    ) private {
+        require(_amount > 0, "OTCWrapper: amount cannot be 0");
+
+        // Sign for transfer approval
+        IERC20Permit(_asset).permit(_acct, address(this), _amount, _deadline, _v, _r, _s);
```

Comment:
we do permit() only if _asset is USDC. Otherwise they need to have approved beforehand. Remember that the __mmSignature might be empty if the collateral asset is not USDC

Reply:
https://github.com/ribbon-finance/GammaProtocol-OTC/pull/2/commits/2f78baa7c4571c2b81b3dc344bf32bd293f52114